scripts/check.sh

```diff
@@ -150,6 +151,22 @@ if [[ "${ALLOW_DIRTY}" == true && "${COMMIT_METHOD}" == "message" ]]; then
   exit 1
 fi
 
+# When travis clones a repo for building, it uses a shallow clone. After the
+# first commit on a non-master branch, TRAVIS_COMMIT_RANGE is not set, master
+# is not available and we need to compute the START_REVISION from the common
+# ancestor of $TRAVIS_COMMIT and origin/master.
+if [[ "${TEST_ONLY}" == true ]]; then
+  if ! [[ -z $TRAVIS_COMMIT_RANGE ]] ; then
```

Comment:
This doesn't work when running under `set -u`, which fails when you access an unset variable. You need to default an unset variable to empty explicitly with a `:-` in braces, as in `${TRAVIS_COMMIT_RANGE:-}`.

`! [[ -z foo ]]` is more idiomatically written as `[[ -n foo ]]`.